### ISSUE TITLE
[RFC] Typography update

### ISSUE THREAD

**Issue by owendodd** [artsy/palette#119]
I recently finalized a proposal the outlines a refinement / evolution of our existing type system. This update would switch our naming conventions from our more abstract / value based system to something that implies the variables usage, with the hopes of increasing efficiency and ease of use. 

This system also incorporates a few other ideas that have previously been discussed. Mainly, the addition of responsive resizing for each variable. There are also additional limitations in regards to available weights and styles at each step to further create consistency in the usage of each. 

I'm looking for general feedback on this system and also any specific thoughts around ways we could potentially implement (whether this acts as a layer on top of our existing primitives or replaces them, etc.)

Specifics can be seen here: 

Zeplin: zpl://screen?sid=5bd3461a2563d758311d772c&pid=5acd19ff49a1429169c3128b
Notion: https://www.notion.so/artsy/Typography-42e0e8e4f4e34ccc84c7de7adcd1b519

**Comment by owendodd** [comment]
@damassi yes, should be accessible now!

**Comment by damassi** [comment]
cc @zephraph

**Comment by damassi** [comment]
@owendodd - just a note that we've written a _lot_ of code using the numeric system and this won't be a pleasant refactor! We'll need to discuss migration strategies.

**Comment by owendodd** [comment]
@damassi yes definitely. Let's discuss at the next meeting if / when / how we would approach this.

**Comment by zephraph** [comment]
Yeah, this was mentioned as a potential risk in the last meeting.

**Comment by jonallured** [comment]
Cool, this is totally what I was hoping for, thanks for writing this up!

I find it easier to talk about proposals with a little sketch to point at so I whipped up this: https://github.com/jonallured/palette/commit/3be6387c0e8ca0675b969e11b27e9a70ea0b95e3.

I took the first three headings and wrote out what I _think_ the implementation would be. Did I get the intention right here? We're basically just formalizing and naming certain family/size/weight/style combos, right? Would we be deprecating customers of palette that are using `Sans` and `Serif` directly? Like, would those end up being "private" at some point?

Regardless, I guess the migration would be some find/replace magic - easier for headings that don't cover both families, but shouldn't be too bad. We could also analyze where we have used combos not covered by this proposal.
